packages/dui3/components/model/Receiver.vue

```diff
@@ -0,0 +1,255 @@
+    await store.receiveModelCancel(props.modelCard.modelCardId)
+  }
+  await store.patchModel(props.modelCard.modelCardId, {
+    selectedVersionId: selectedVersion.id,
+    latestVersionId: latestVersion.id, // patch this dude as well, to make sure
+    hasSelectedOldVersion: selectedVersion.id === latestVersion.id
+  })
+
+  await store.receiveModel(props.modelCard.modelCardId)
+}
+
+// Cancels any in progress receive OR receives latest version
+const handleMainButtonClick = async () => {
+  if (props.modelCard.progress)
+    return await store.receiveModelCancel(props.modelCard.modelCardId)
+  await receiveLatestVersion()
+}
+
+// Cancels any in progress receive AND receives latest version
+const receiveLatestVersion = async () => {
+  // Note: here we're updating the model card info, and afterwards we're hitting the receive action
+  await store.patchModel(props.modelCard.modelCardId, {
+    selectedVersionId: props.modelCard.latestVersionId
+  })
+  if (props.modelCard.progress)
+    await store.receiveModelCancel(props.modelCard.modelCardId)
+  await store.receiveModel(props.modelCard.modelCardId)
+}
+
+const expiredNotification = computed(() => {
+  if (!props.modelCard.latestVersionId || props.modelCard.hasDismissedUpdateWarning)
+    return
+  if (props.modelCard.latestVersionId === props.modelCard.selectedVersionId) return
+  const notification = {} as ModelCardNotification
+  notification.dismissible = true
+  notification.level = 'warning'
+  notification.text = 'Newer version available!'
+  notification.cta = {
+    name: 'Update',
+    action: receiveLatestVersion
+  }
+  return notification
+})
+
+const receiveResultNotification = computed(() => {
+  if (
+    !props.modelCard.bakedObjectIds ||
+    props.modelCard.displayReceiveComplete !== true
+  )
+    return
+
+  const notification = {} as ModelCardNotification
+  notification.dismissible = true
+  notification.level = 'success'
+  notification.text = 'Model loaded!'
+  notification.report = props.modelCard.report
+  notification.cta = {
+    name: 'Highlight',
+    action: () => {
+      app.$baseBinding.highlightModel(props.modelCard.modelCardId)
+    }
+  }
+  return notification
+})
+
+const errorNotification = computed(() => {
+  if (!props.modelCard.error) return
+  const notification = {} as ModelCardNotification
+  notification.dismissible = true
+  notification.level = 'danger'
+  notification.text = props.modelCard.error
+  notification.report = props.modelCard.report
+  return notification
+})
+
+const { result: versionDetailsResult, refetch } = useQuery(
+  versionDetailsQuery,
+  () => ({
+    projectId: props.modelCard.projectId,
+    modelId: props.modelCard.modelId,
+    versionId: props.modelCard.selectedVersionId
+  }),
+  () => ({
+    clientId: props.modelCard.accountId
+  })
+)
+
+const createdAgoUpdater = useInterval(10_000) // refresh the created ago, and latestversion etc. every 10s
+
+const createdAgo = computed(() => {
+  createdAgoUpdater.value
```

Comment:
I guess this works, but overly clever & confusing IMO. I suggest using a reactive now date like we use in FE2 instead.

```ts
const now = useReactiveNowDate({updateEvery: 10_000})
const createdAgo = computed(() => {
  return dayjs(versionDetailsResult.value?.project.model.version.createdAt).from(now.value)
})
```

Same goes for other functions that do this